code/math/implementation.lisp

```diff
@@ -38,15 +189,101 @@
          (ash p ,(- size)))))
 
 (defun round-to-odd/32 (g cp)
-  #-(or ecl cmucl) (%round-to-odd-1 g cp 32)
-  #+(or ecl cmucl) (%round-to-odd-2 g cp 32))
+  #+quaviver/bignum-elision
+  (let ((p (*/32-64/hi64 cp g)))
+    (if (ldb-test (byte 31 1) p)
+        (logior (ash p -32) 1)
+        (ash p -32)))
+  #-quaviver/bignum-elision
+  (progn
+    #-(or ecl cmucl) (%round-to-odd-1 g cp 32)
```

Comment:
I'd ditch the `progn` and just do 
```
 #-(and (not quaviver/bignum-elision) (or ecl cmucl)) (%round-to-odd-1 g cp 32)
 #+(and (not quaviver/bignum-elision) (or ecl cmucl)) (%round-to-odd-2 g cp 32)
```

Yes, its a little more verbose, but it puts all of the conditional expressions on the same level and is bit clearer.